lib/web_ui/lib/src/engine/text_editing/text_editing.dart

```diff
@@ -1027,13 +1174,52 @@ abstract class DefaultTextEditingStrategy implements TextEditingStrategy {
     assert(isEnabled);
 
     final EditingState newEditingState = EditingState.fromDomElement(activeDomElement);
+    TextEditingDeltaState? newTextEditingDeltaState;
+    if (inputConfiguration.enableDeltaModel) {
+      newTextEditingDeltaState = TextEditingDeltaState.inferDeltaState(newEditingState, lastEditingState, editingDelta);
```

Comment:
Related to a comment I left above about modifying function arguments: Is editingDelta equivalent to newTextEditingDeltaState after this line?  If so, it would simplify things to not modify the function parameter and just modify a copy, and then to use newTextEditingDeltaState exclusively after this.

Reply:
editingDelta is not equivalent to newTextEditingDeltaState after inferDeltaState completes. iirc I don't think dart supports pass by reference. I agree making a copy instead would simplify things here.